src/RequestExecutor/HttpRequestExecutor.spec.ts

```diff
@@ -23,71 +16,57 @@ import {
   verify,
   when
 } from 'ts-mockito';
-import { container, Lifecycle } from 'tsyringe';
 import { URL } from 'url';
 
-describe('HttpRequestExecutor', () => {
-  const createRequest = (options?: Partial<RequestOptions>) => {
-    const requestOptions = { url: 'https://foo.bar', headers: {}, ...options };
-    const request = new Request(requestOptions);
-    const spiedRequest = spy(request);
-    when(spiedRequest.method).thenReturn('GET');
+const createRequest = (options?: Partial<RequestOptions>) => {
+  const requestOptions = { url: 'https://foo.bar', headers: {}, ...options };
+  const request = new Request(requestOptions);
+  const spiedRequest = spy(request);
+  when(spiedRequest.method).thenReturn('GET');
 
-    return { requestOptions, request, spiedRequest };
-  };
+  return { requestOptions, request, spiedRequest };
+};
 
-  const virtualScriptsMock = mock<VirtualScripts>(DefaultVirtualScripts);
+describe('HttpRequestExecutor', () => {
+  const virtualScriptsMock = mock<VirtualScripts>();
+  const executorOptions: RequestExecutorOptions = {};
+  const spiedExecutorOptions = spy(executorOptions);
```

Comment:
Spy just wraps an existing instance, it will still behave in the same way as a regular object. That allows you to stub (configure/override) the behavior of its instance members without monkey patching of global variables. Do not confuse with mock.

Reply:
Yeah, I mean it is all OK, for sure. Just to clarify: according to [this classification](https://martinfowler.com/bliki/TestDouble.html) a `spy` is something that **records** it's invocations and we don't need it here (and don't use BTW) so a **stub** would suffice just fine. And since `executorOptions` is a plain object there is no difference at all between stubbing it or just using a brand new plain object for each test case.

Reply:
You are right, there's no difference between them in this particular case. You can create a SUT during a test invocation. But not only that makes code more complex and uncertain, also prospects for expanding POJO to something more. Referencing to Martin's definition, you should also be made aware of the spy is used for real objects, when stub is not. Anyway, let's leave this discussion until I get back